### PR TITLE
Fix cluster group handling on instance creation

### PR DESCRIPTION
**Pull request by stgraber**

### CODE REVIEW

File: lxd/db/node.go [hunk 1]
```diff
@@ -1169,13 +1168,13 @@ func (c *ClusterTx) GetNodeWithLeastInstances(archs []int, defaultArch int, grou
...
 
-		// Fetch the number of containers already created on this node.
+		// Fetch the number of instances already created on this node.
 		created, err := query.Count(c.tx, "instances", "node_id=?", node.ID)
 		if err != nil {
 			return "", errors.Wrap(err, "Failed to get instances count")
 		}
 
-		// Fetch the number of containers currently being created on this node.
+		// Fetch the number of isntances currently being created on this node.
```
Comment (tomponline) [inline lxd/db/node.go:1192]: Typo

Comment (stgraber) [reply lxd/db/node.go]: Fixed

File: lxd/instances_post.go [hunk 2]
```diff
@@ -859,6 +860,14 @@ func instancesPost(d *Daemon, r *http.Request) response.Response {
 			return response.SmartError(err)
 		}
 
+		// Load restricted groups from project.
+		var allowedGroups []string
+		if !isClusterNotification(r) && shared.IsTrue(p.Config["restricted"]) {
+			allowedGroups = util.SplitNTrimSpace(p.Config["restricted.cluster.groups"], ",", -1, true)
+		} else {
+			allowedGroups = nil
```
Comment (tomponline) [inline lxd/instances_post.go:868]: No need for this line as will be a nil slice from line 864

**Review by tomponline — APPROVED** [review]

**Comment by stgraber** [comment]
@tomponline good to go

**Pull request merged**